packages/typespec-ts/test/integration/modelInheritanceEnumDiscriminator.spec.ts

```diff
@@ -0,0 +1,126 @@
+import EnumDiscriminatorClientFactory, {
+  EnumDiscriminatorClient,
+  Golden,
+  Snake
+} from "./generated/models/inheritance-enum-discriminator/src/index.js";
+import { assert } from "chai";
+
+describe("EnumDiscriminatorClient Rest Client", () => {
+  let client: EnumDiscriminatorClient;
+
+  beforeEach(() => {
+    client = EnumDiscriminatorClientFactory({
+      allowInsecureConnection: true
+    });
+  });
+
+  const validBody: Golden = {
+    weight: 10,
+    kind: "golden"
+  };
+  const validFixedEnumBody: Snake = {
+    length: 10,
+    kind: "cobra",
+  };
+  it("should get extensible enum", async () => {
+    try {
+      const result = await client
+        .path("/type/model/inheritance/enum-discriminator/extensible-enum")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, validBody);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put extensible enum", async () => {
+    try {
+      const result = await client
+        .path("/type/model/inheritance/enum-discriminator/extensible-enum")
+        .put({ body: validBody });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get extensible enum if missing discriminator", async () => {
+    try {
+      const result = await client
+        .path(
+          "/type/model/inheritance/enum-discriminator/extensible-enum/missingdiscriminator"
+        )
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, { weight: 10 } as any);
```

Comment:
nit: should remove `as any`?
```suggestion
      assert.deepEqual(result.body, { weight: 10 });
```